### DISCUSSION TITLE
Make `go` installation optional for `golangci-lint`

### DISCUSSION THREAD

**Discussion by antklim** [Homebrew/discussions#3925] [Everyday usage] [answered]
Hi homebrew team,

I'm not sure what category the question belongs to, therefore asking it in the general category. 

I'm wondering if it's possible to update the `golangci-lint` installation process and not install go if it's already present. What happens now is:

- I have `go` installed on my machine. I used an installer from https://go.dev/dl/.
- I install `golangci-lint` using brew. It implicitly installs `go` using brew. As a result, I have two sources of `go` installed on my machine. Eventually, the versions of different `go` start drifting.

Thanks,
Anton

**Comment by SMillerDev** [comment, marked as answer]
No, Homebrew formula with dependencies install those dependencies through Homebrew

**Reply by antklim** [reply]
Got it. Thanks for your reply.

**Comment by informatik01** [comment]
@antklim 
To avoid installing Go as a dependency for _golangci-lint_ you can use the dedicated [_tap_](https://docs.brew.sh/Taps) (i.e. a third-party repository) provided by the authors of _galangci-lint_.

Recently I was tired of waiting for Go version 1.20 to appear in the official Homebrew [_formulae_](https://docs.brew.sh/Formula-Cookbook), so I installed the latest version of Go from the Go [website](https://go.dev/dl/). And in order to avoid installing another (in this case older - 1.19.6) version of Go as the _golangci-lint_ dependency, I added the dedicated _tap_ and installed _golangci-lint_ form there.

---
Here is a quote from the official golangci-lint [documentation](https://golangci-lint.run/usage/install/#macos):
> Previously we used a [homebrew tap](https://github.com/golangci/homebrew-tap). We recommend using official formula instead of the tap, but **sometimes the most recent release isn't immediately available via homebrew core** due to manual updates that need to occur from homebrew core maintainers. In this case, the tap formula, which is updated automatically, can be used to install the latest version of golangci-lint:
```sh
brew tap golangci/tap
brew install golangci/tap/golangci-lint
```

Now I have both the latest version of Go (1.20.1) which was installed manually (i.e. not using Homebrew) and the latest version of _golangci-lint_ installed using Homebrew 😀

**Reply by antklim** [reply]
Awesome! Thanks for the advice @informatik01! 👍

**Reply by informatik01** [reply]
@antklim You're welcome!
